File: Python x EDAII/Ordenamiento/PigeonholeSort.py
class PigeonholeSort:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase'
		self.arr = arr

	def pigeonholeSort(self):
		'Algoritmo de ordenamiento.'
		maximo = max(self.arr)
		minimo = min(self.arr)
		rango = maximo-minimo+1
		conteo = [0] * rango	# Arreglo para conteo

		# Conteo de la cantidad de elementos hacia su respectivo casillero
		for i in range(0,len(self.arr)):
			conteo[self.arr[i]-minimo]+=1

		# Reasignando elementos de manera ordenada
		i = 0
		for cuenta in range(0,rango):
			while conteo[cuenta] > 0:
				conteo[cuenta] -= 1
				self.arr[i] = cuenta + minimo
				i = i+1

# Ejecutando el objeto
programa = PigeonholeSort([55,63,61,59,63,57,57,59,58,60])
programa.pigeonholeSort()
print(programa.arr)

File: EDAII x Java/Ordenamiento/PigeonholeSort.java
import java.util.*;

public class PigeonholeSort{

	int i;
	int[] arr = {55,63,61,59,63,57,57,59,58,60};

	public static void main(String[] args){
		PigeonholeSort ps = new PigeonholeSort();
		ps.pigeonholeSort(ps.arr);  //Ordenamiento PigeonholeSort

		for(ps.i=0; ps.i<=ps.arr.length-1; ps.i++){
			System.out.print(ps.arr[ps.i] + ", ");
		}
		System.out.println("\n");
	}

	public void pigeonholeSort(int[] A){
		int max = Arrays.stream(A).max().getAsInt();
		int min = Arrays.stream(A).min().getAsInt();
		int rango = max-min+1;
		int conteo[] = new int[rango]; // Arreglo para conteo

		// Conteo de la cantidad de elementos hacia su respectivo casillero
		for (int j=0; j<A.length; j++){
			conteo[A[j]-min]++;
		}

		// Reasignando elementos de manera ordenada
		i = 0;
		for (int cuenta=0; cuenta<rango; cuenta++){
			while (conteo[cuenta] > 0){
				conteo[cuenta] -= 1;
				A[i] = cuenta + min;
				i = i+1;
			}
		}
	}

}